### PR TITLE
feat: added new page templates

### PR DESCRIPTION
**Pull request by FionnD**
Added new templates based off other open-source companies approach.

### CODE REVIEW

File: page_templates/developer_guide_how_to.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
+How To Template (Use Title Case)
+============================================================
+
+Section headers use sentence case
+
```
Comment (bwanglzu) [inline page_templates/developer_guide_how_to.rst:5]: maybe we can add `meta` field here for seo purpose, and table of contents.

File: page_templates/overview.rst [hunk 2]
```diff
@@ -1,3 +1,2 @@
-=================
-Short title
+Short Title (Use Title Case)
 =================
```
Comment (bwanglzu) [inline page_templates/overview.rst]: make this line longer

**Review by bwanglzu — CHANGES_REQUESTED** [review]

File: page_templates/developer_guide_how_to.rst (hunk 1 above)
Comment (FionnD) [reply page_templates/developer_guide_how_to.rst:5]: Hmm yes good point, but should we have a predefined list of keywords to choose from? or people create their own?

Comment (FionnD) [reply page_templates/developer_guide_how_to.rst:5]: I'm not so sure about table of contents. Ideally, the page should be short enough not to have one :)

Comment (FionnD) [reply page_templates/developer_guide_how_to.rst:5]: You mean adding this? https://docs.readthedocs.io/en/stable/guides/technical-docs-seo-guide.html#use-meta-tags

Comment (bwanglzu) [reply page_templates/developer_guide_how_to.rst:5]: yes, we can also add `keywords` to `meta`

Comment (FionnD) [reply page_templates/developer_guide_how_to.rst:5]: I think these are good ideas, but maybe it is too much for the moment? I don't want to have developers do to much work

File: page_templates/overview.rst (hunk 2 above)
Comment (FionnD) [reply page_templates/overview.rst]: Done

Comment (FionnD) [reply page_templates/overview.rst]: Done

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**